File: Stage12/Java/Problem2231.java
package Stage12.Java;

import java.util.Scanner;

public class Problem2231 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int result = 0;
        int start = N - 9 * String.valueOf(N).length();
        if (start < 1) start = 1;

        for (int i = start; i < N; i++) {
            int sum = i;
            int temp = i;

            while (temp > 0) {
                sum += temp % 10;
                temp /= 10;
            }

            if (sum == N) {
                result = i;
                break;
            }
        }

        System.out.println(result);
        sc.close();
    }
}

File: Stage12/Python/Problem2231.py
def main():
    N = int(input())
    start = max(1, N - 9 * len(str(N)))
    result = 0
    
    for i in range(start, N):
        total = i + sum(map(int, str(i)))
        if total == N:
            result = i
            break
    
    print(result)

if __name__ == "__main__":
    main()